### ISSUE TITLE
[BUG] Fix the sync of migrate button on UI

### ISSUE THREAD

**Issue by w13915984028** [harvester/harvester#7180]
**Describe the bug**

Harvester render the UI `migrate` button via check vmi object annotation.

https://github.com/harvester/harvester/blob/a9cfd042a3b707108ff6337a933dc646bfbde64a/pkg/api/vm/formatter.go#L106

```
	if canMigrate(vmi) {
		resource.AddAction(request, migrate)
		resource.AddAction(request, findMigratableNodes)
	}
```

And, the vmim controller checks the vmim object then update the vmi object annotation.

https://github.com/harvester/harvester/blob/a9cfd042a3b707108ff6337a933dc646bfbde64a/pkg/controller/master/migration/vmim_controller.go#L45

```
...
	if vmim.Status.Phase == kubevirtv1.MigrationPending && !abortRequested {
		return vmim, h.scaleResourceQuota(vmi)
	} else if vmim.Status.Phase != kubevirtv1.MigrationFailed && abortRequested {
		return vmim, h.setVmiMigrationUIDAnnotation(vmi, string(vmim.UID), StateAbortingMigration)
	}
...
```

You may note that, when `vmim.Status.Phase == kubevirtv1.MigrationPending`, the `vmi` object is not update but the `resourcequota` object  (refer another issue: https://github.com/harvester/harvester/issues/7178)

In this phase, the UI still shows the `migrate` button, if user tries to trigger the migrate from UI, there is such error:

![image](https://github.com/user-attachments/assets/2bafc29d-9119-4e88-af43-93d2e8bd1733)

kubevirt denied the migration, as the `vmim` object is already there.

**To Reproduce**
1. Create 4 or more VMs on a same node
2. Maintain the node
3. 4 VM migrations are created
4. 2 of them starts ( by default 2 in parallel)
5. check UI, another 2 VMs sitll show `migrate` button, if you click it, the related error will be reported

**Expected behavior**

Render the UI properly

**Support bundle**

**Environment**
 - Harvester ISO version:  Harvester v1.3.1
 - Underlying Infrastructure (e.g. Baremetal with Dell PowerEdge R630):

**Additional context**

**Comment by bk201** [comment]
@w13915984028 please add the require-ui labels if need UI help or change to `area/ui` if this is purely a UI issue.

**Comment by w13915984028** [comment]
@bk201 At the moment, this is mainly a backend issue.

**Comment by w13915984028** [comment]
It looks the fix PR https://github.com/harvester/harvester/pull/7195 won't be a simple and easy one.

The status is synced between vm, vmi, vmim objects, and UI relies on the VM object to be synced and update the UI, the current code has some mis-matchings.

For anyone who has interests, please leave your comments on the PR, thanks.

**Comment by Vicente-Cheng** [comment]
Hi @w13915984028, 
Could you share the status of this issue? Thanks!

**Comment by w13915984028** [comment]
**test steps** are on PR desc: https://github.com/harvester/harvester/pull/8711

**Comment by asc5543** [comment]
## Test result
Test PASS

## Environment
* Harvester ISO version: `v1.6-429e841a-head`
* Underlying Infrastructure (e.g. Baremetal with Dell PowerEdge R630): `2 node via ipxe-example`


## Test Scenario
### Preparation
1. <details> <summary> Create 4 or more VMs on a same node </summary>

   <img width="1908" height="987" alt="Image" src="https://github.com/user-attachments/assets/7502bc41-d0e4-491b-8df7-292e74ac9035" />

   </details>


### Test Steps
1. <details>
   <summary> ✔ Maintain the node
   </summary>

   <img width="1908" height="987" alt="Image" src="https://github.com/user-attachments/assets/418e8532-bcbc-474e-b8bc-0bc8e3a2c49c" />

   </details>

1. <details>
   <summary> ✔ 4 VM migrations are created
   </summary>
  
   <img width="1908" height="987" alt="Image" src="https://github.com/user-attachments/assets/443a1246-75b9-4b93-aeb0-774de2775d9a" />

   </details>

1. <details>
   <summary> ✔ 2 of them starts ( by default 2 in parallel)
   </summary>

   <img width="1908" height="987" alt="Image" src="https://github.com/user-attachments/assets/d330d400-c083-4342-9e11-8cc1ffe188c9" />

   </details>

1. <details>
   <summary> ✔ Check UI, the still migrating vms have UI menu Abort Migration
   </summary>

   <img width="1908" height="987" alt="Image" src="https://github.com/user-attachments/assets/39339fa4-215d-4ad2-b470-b3d236101cf0" />
   <img width="1908" height="987" alt="Image" src="https://github.com/user-attachments/assets/0e694320-2d1a-4ec0-b440-12f29124fcea" />

   </details>

1. <details>
   <summary> ✔ Click migrate on a vm
   </summary>

   <img width="1910" height="987" alt="Image" src="https://github.com/user-attachments/assets/ee5bb0bc-eb0d-4cf7-ae28-f90786dda272" />
   <img width="1910" height="987" alt="Image" src="https://github.com/user-attachments/assets/ca261894-4336-44dc-b59a-c1451dc913ec" />

   </details>

1. <details>
   <summary> ✔ When a vm is in migrating/pending, you can click Abort Migration
   </summary>

   <img width="1910" height="987" alt="Image" src="https://github.com/user-attachments/assets/2bd3aa17-d2bd-4ca1-9a3d-c861ef01139c" />
   <img width="1910" height="987" alt="Image" src="https://github.com/user-attachments/assets/42536f39-c958-4d70-bb38-150ea2acb9d8" />

   </details>
